File: backend/FinancialControl.Api/Models/Invitation.cs
namespace FinancialControl.Api.Models;

public class Invitation
{
    public Guid Id { get; set; }
    public Guid AccountId { get; set; }
    public Guid InvitedByUserId { get; set; }
    public string InvitedEmail { get; set; } = string.Empty;
    public AccountRole Role { get; set; }
    public InvitationStatus Status { get; set; }
    public string Token { get; set; } = string.Empty;
    public DateTime ExpiresAt { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? AcceptedAt { get; set; }

    // Navigation properties
    public Account Account { get; set; } = null!;
    public User InvitedByUser { get; set; } = null!;
}

public enum InvitationStatus
{
    Pending = 1,
    Accepted = 2,
    Rejected = 3,
    Expired = 4,
    Cancelled = 5
}

File: frontend/src/types/invitation.ts
export type AccountRole = 'Owner' | 'Editor' | 'Viewer';

export type InvitationStatus = 'Pending' | 'Accepted' | 'Rejected' | 'Expired' | 'Cancelled';

export interface Invitation {
  id: string;
  accountId: string;
  accountName: string;
  invitedByUserId: string;
  invitedByUserName: string;
  invitedEmail: string;
  role: AccountRole;
  status: InvitationStatus;
  expiresAt: string;
  createdAt: string;
  acceptedAt?: string;
}

export interface InvitationListItem {
  id: string;
  invitedEmail: string;
  role: AccountRole;
  status: InvitationStatus;
  expiresAt: string;
  createdAt: string;
  invitedByUserName: string;
}

export interface CreateInvitationRequest {
  accountId: string;
  invitedEmail: string;
  role: AccountRole;
}

export interface AcceptInvitationRequest {
  token: string;
}

export const ROLE_LABELS: Record<AccountRole, string> = {
  Owner: 'Proprietário',
  Editor: 'Editor',
  Viewer: 'Visualizador',
};

export const ROLE_DESCRIPTIONS: Record<AccountRole, string> = {
  Owner: 'Controle total da conta',
  Editor: 'Pode adicionar e editar transações',
  Viewer: 'Apenas visualizar',
};

export const STATUS_LABELS: Record<InvitationStatus, string> = {
  Pending: 'Pendente',
  Accepted: 'Aceito',
  Rejected: 'Rejeitado',
  Expired: 'Expirado',
  Cancelled: 'Cancelado',
};
